src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -119,25 +132,51 @@ public void SettingUpdatesPropertiesOnSymlink(bool targetExists)
             T link = CreateSymlinkToItem(target);
             if (!targetExists)
             {
-                SettingUpdatesPropertiesCore(link);
+                if (!SettingPropertiesUpdatesLink)
+                {
+                    SettingUpdatesPropertiesCore(target, link);
+                }
             }
             else
             {
                 // Get the target's initial times
                 IEnumerable<TimeFunction> timeFunctions = TimeFunctions(requiresRoundtripping: true);
-                DateTime[] initialTimes = timeFunctions.Select((funcs) => funcs.Getter(target)).ToArray();
+                DateTime[] initialTimes = timeFunctions
+                    .Select((func) => func.Getter(target))
+                    .ToArray();
 
-                SettingUpdatesPropertiesCore(link);
+                SettingUpdatesPropertiesCore(target, link);
 
                 // Ensure that we have the latest times.
                 if (target is FileSystemInfo fsi)
                 {
                     fsi.Refresh();
                 }
 
+                T getTarget = SettingPropertiesUpdatesLink ? link : target;
+
                 // Ensure the target's times haven't changed.
-                DateTime[] updatedTimes = timeFunctions.Select((funcs) => funcs.Getter(target)).ToArray();
-                Assert.Equal(initialTimes, updatedTimes);
+                DateTime[] updatedTimes = timeFunctions
+                    .Select(func => func.Getter(getTarget))
+                    .ToArray();
+                if (SettingPropertiesUpdatesLink)
+                {
+                    for (int i = 0; i < initialTimes.Length; i++)
+                    {
+                        DateTime initialTime = initialTimes[i];
+                        DateTime updatedTime = updatedTimes[i];
+
+                        // There is some precision loss between initialTimes and updatedTimes,
+                        // so we check for a delta of max 1.5 seconds (15,000,000 ticks)
+                        Assert.True(
+                            Math.Abs(updatedTime.Ticks - initialTime.Ticks) < 1.5 * TimeSpan.TicksPerSecond,
+                            $"{nameof(updatedTime)} ({updatedTime}) is not similar to {nameof(initialTime)} ({initialTime})");
+                    }
+                }
+                else
+                {
+                    Assert.NotEqual(initialTimes, updatedTimes);
+                }
```

Comment:
```suggestion
                DateTime[] updatedTimes = timeFunctions.Select((funcs) => funcs.Getter(getTarget)).ToArray();
                Assert.Equal(initialTimes, updatedTimes);
```

Reply:
This verifies `getTarget` did not change.

Reply:
@tmds This change is causing test failures. Suggestions?

```
      System.IO.Tests.File_GetSetTimes_SafeFileHandle.SettingPropertiesOnSymlink(targetExists: True) [FAIL]
        Assert.Equal() Failure
                   ↓ (pos 0)
        Expected: [2022-05-02T19:43:59.4862214+02:00, 2022-0···
        Actual:   [2014-12-01T13:03:03.3210000+01:00, 2014-1···
                   ↑ (pos 0)
```

```cs
        [ConditionalTheory(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
        [PlatformSpecific(~TestPlatforms.Browser)] // Browser is excluded as it doesn't support symlinks
        [InlineData(false)]
        [InlineData(true)]
        public void SettingPropertiesOnSymlink(bool targetExists)
        {
            // This test is in this class since it needs all of the time functions.
            // This test makes sure that the times are set on the symlink itself.
            // It is needed as on OSX for example, the default for most APIs is
            // to follow the symlink to completion and set the time on that entry
            // instead (eg. the setattrlist will do this without the flag set).
            // It is also the same case on unix, with the utimensat function.
            // It is a theory since we test both the target existing and missing.

            T target = targetExists ? GetExistingItem() : GetMissingItem();

            // When the target exists, we want to verify that its times don't change.

            T link = CreateSymlinkToItem(target);
            if (!targetExists)
            {
                if (SettingPropertiesUpdatesLink)
                {
                    SettingUpdatesPropertiesCore(link, target);
                }
            }
            else
            {
                // Get the target's initial times
                IEnumerable<TimeFunction> timeFunctions = TimeFunctions(requiresRoundtripping: true);
                DateTime[] initialTimes = timeFunctions
                    .Select((func) => func.Getter(target))
                    .ToArray();

                SettingUpdatesPropertiesCore(link, target);

                // Ensure that we have the latest times.
                if (target is FileSystemInfo fsi)
                {
                    fsi.Refresh();
                }

                T getTarget = SettingPropertiesUpdatesLink ? link : target;

                // Ensure the target's times haven't changed.
                DateTime[] updatedTimes = timeFunctions.Select((funcs) => funcs.Getter(getTarget)).ToArray();
                Assert.Equal(initialTimes, updatedTimes);
            }
        }
 ```

Reply:
ok, that makes sense. We end up pulling the target properties always when !SettingPropertiesUpdatesLink, so we can't verify the link properties haven't changed. They may change anyway because we're going to access the link to get to the target. So let's skip them.

Can you cherry-pick: https://github.com/tmds/runtime/commit/b82332a4d251e0adf9a07bd2037d476faeb4a471